joinmarket-webui/jam-docker
# feat(init): use dinit instead of supervisor

Resolves #62.

Before this PR, there was a short delay (5s) when starting `jmwalletd` to wait for tor port bindings, as the previous service manager (`supervisord`) has no concept of dependencies. This is brittle and error-prone.

After this PR is applied, `supervisord` has been replaced by `dinit` which can build [service hierarchies](https://github.com/davmac314/dinit#service-description-files) and making it possible for `jmwalletd` and `ob-watcher` to actively wait for `tor` to start successfully.

Also updates docker image versions:
- build image: `alpine` from `3.15` to `3.16.2`
- runtime image `debian` from `bullseye-20220801-slim` to `bullseye-20221004-slim`

### How to test?
Use the image published from this branch called `dinit` in your regtest environment by changing label `master` to `dinit` in [dockerfile-deps/joinmarket/webui-standalone/Dockerfile](https://github.com/joinmarket-webui/jam/blob/master/docker/regtest/dockerfile-deps/joinmarket/webui-standalone/Dockerfile#L1) and rebuild your setup with `npm run regtest:build`.
You should see instance 2 and 3 come up normally and access their UI on http://localhost:29080 and http://localhost:30080.
In the console you see smth like:
```
jm_regtest_joinmarket2          | [  OK  ] nginx
jm_regtest_joinmarket2          | [  OK  ] tor
jm_regtest_joinmarket2          | [  OK  ] jmwalletd
jm_regtest_joinmarket2          | [  OK  ] ob-watcher
jm_regtest_joinmarket2          | [  OK  ] boot
```
